processing/src/main/java/org/apache/druid/segment/virtual/ExpressionVirtualColumn.java

```diff
@@ -62,6 +64,23 @@ public ExpressionVirtualColumn(
     this.parsedExpression = Suppliers.memoize(() -> Parser.parse(expression, macroTable));
   }
 
+  /**
+   * Constructor for creating an ExpressionVirtualColumn from a pre-parsed expression.
+   */
+  public ExpressionVirtualColumn(
+      String name,
+      Expr parsedExpression,
+      ValueType outputType
+  )
+  {
+    this.name = Preconditions.checkNotNull(name, "name");
+    // Unfortunately this string representation can't be reparsed into the same expression, might be useful
+    // if the expression system supported that
+    this.expression = parsedExpression.toString();
```

Comment:
This code is nasty for the reason you mentioned in your comment. I don't think it'd be good to leave it like this, since it means any attempt to serialize this virtual column will yield something unusable, potentially leading to latent bugs.

I'd almost consider _not_ supporting pushdown of filters where the lhs is an expression until we have a way to actually convert parsed expressions to strings.

Another option:

1) Make a single private constructor of ExpressionVirtualColumn that takes a nullable 'expression' and an optional pre-parsed expression.
2) Make a `@JsonCreator` static factory method that Jackson will use.
3) Make another static factory method that accepts a pre-parsed expression, and stores null for the 'expression'.
4) Make `getExpression()` throw an exception if 'expression' is null. This means anyone trying to serialize it will realize what is going on before getting hit by latent bugs.
5) Make sure to note in the javadoc of JoinFilterAnalysis that the virtual columns are not serializable.

Open to other ideas.

Reply:
I decided to just disable lhs expression push down for now (JoinFilterColumnCorrelationAnalysis.supportsPushDown returns false now if there are lhs expressions), but kept the rest of the code there and added an `@Ignore` annotation to tests that have lhs expressions (since the "only" thing blocking it is the lack of support from the expression system)

Reply:
I think this is a reasonable course of action. Could you please also raise a followup github issue.

Reply:
I opened https://github.com/apache/druid/issues/9326 to track the expressions enhancement and mentioned it in a comment here